lib/facil/http/http.c

```diff
@@ -2073,8 +2073,12 @@ struct tm *http_gmtime(time_t timer, struct tm *tmbuf) {
   if (timer < 0)
     return gmtime_r(&timer, tmbuf);
   ssize_t a, b;
+#ifdef HAVE_TM_TM_GMTOFF
```

Comment:
How can we add compile-time detection for these flags in cases where the `makefile` isn't used?

Reply:
I can't find a macro in `sys/time.h` that specifically indicate those additional fields are available, from either FreeBSD or GNU (GLIBC).

The GLIBC header enables those fields based on macro `__GNU_MISC`, I think it was controlled by `_GNU_SOURCE`.
While the FreeBSD header have those fields unconditionally.

Reply:
Just seen `sys/time.h` in OS X and OpenBSD; it seems that all BSDs has those fields. 
ctime(3) for GLIBC said those 2 fields will be available if `_BSD_SOURCE` is defined.

I can add test macros for all known BSD systems, but testing on GNU system is somewhat harder; because in my test it will also be available based on some other macros, such as `_SVID_SOURCE`, in this case `_BSD_SOURCE` won't be defined by the GLIBC headers.

Reply:
I decide to test macro `_BSD_SOURCE` on GNU systems, just based on what its man page said.

Reply:
I doubt if `_BSD_SOURCE` is defined on all BSD systems. it's a `glibc` flag often set by the developer on Linux systems.

How about testing for either `__GLIBC__` or and of the Marcos [here](https://github.com/boazsegev/facil.io/blob/ed5998968bfb8d1423a9597d758316cd9ad42fe7/lib/facil/fio.c#L47-L48) as suggested [in this SO answer](https://stackoverflow.com/a/45107992/4025095)...?